src/storage/src/render/persist_sink.rs

```diff
@@ -234,6 +247,11 @@ where
                                 .expect("cannot append updates")
                                 .expect("invalid/outdated upper");
 
+                            source_statistics.inc_messages_committed_by(
+                                batch_builder.inserts + batch_builder.retractions,
+                            );
+                            source_statistics.update_snapshot_committed(&new_upper);
```

Comment:
Is this the race you mentioned, and the reason for the boolean in the `Rc`? Because we might otherwise wrongly report `snapshot_committed == false` on a restart?

Not sure I like it, but also don't have a better idea off the top of my head... 😅

Reply:
yeah we need to actually startup the `persist_sink`~